src/SAML2/XML/md/AbstractIndexedEndpointType.php

```diff
@@ -87,7 +87,7 @@ public static function fromXML(DOMElement $xml): object
      */
     public function toXML(DOMElement $parent = null): DOMElement
     {
-        Assert::notEmpty($this->index);
+        Assert::numeric($this->index);
```

Comment:
May I suggest s/numeric/natural ?

Reply:
You may, but we can't use naturals there 😛

Naturals don't include 0, but that's a valid index number, AFAIU.

Reply:
0 is included in webmozart's natural assertion 😉

Reply:
Uhm, didn't know that webmozart behaves like that, but I just found (to my surprise) that some people (even standards) consider naturals to start at 0... 😮

In any case, agreed, the test should be in the setter. I'll fix it.

Reply:
Yeah, apparently there is no general consensus on whether 0 is a natural number or not, although in most cases it is excluded.. Webmozart specifically defines it as 'non-negative'..